Use library for drop modal positioning, fix story

b12f opened the pull request:
After out-of-tree rendering was implemented for drop-modals, we had some issues surrounding positioning. By using the `@floating-ui/dom` library, we can avoid implementing this logic ourselves, and also fix an overflow scroll bug.

At the same time, this commit fixes the drop-modal story that broke.

Closes https://community.openproject.org/work_packages/46164/activity

Commit:
Use library for drop modal positioning, fix story

After out-of-tree rendering was implemented for drop-modals, we had
some issues surrounding positioning. By using the `@floating-ui/dom`
library, we can avoid implementing this logic ourselves, and also fix
an overflow scroll bug.

At the same time, this commit fixes the drop-modal story that broke.

Closes https://community.openproject.org/work_packages/46164/activity

Commit:
Merge branch 'dev' into fix/46164-drop-modal-opening

Commit:
Merge branch 'dev' into fix/46164-drop-modal-opening

oliverguenther commented:
- [ ] The project dropdown is wrongly positioned:

![image](https://user-images.githubusercontent.com/459462/218678243-c7a37681-20bf-4ea9-b1ab-92da01ecb9c4.png)

oliverguenther reviewed (changes requested):
Looking good :+1: The project dropdown is broken, I couldn't find any other positioning errors

Commit:
Fix header project select opening direction

After switching to floating-ui/dom, the alignment naming has changed.
This commit fixes the name of the alignment, as well as some eslint errors.

Commit:
Fix eslint errors
